addons/godot-package-manager/godot_package_manager.gd

```
@@ -446,722 +849,486 @@ const ValidHooks := {
-		res = _save_data(downloaded_file, download_location)
-		if not res or res.is_err():
-			failed_packages.add(package_name, res.unwrap_err().to_string() if res else DEFAULT_ERROR)
-			continue
+	emit_signal("operation_started", "update", pkg_configs.size())
 
-		#endregion
-
-		if not dir.dir_exists(dir_name):
-			if dir.make_dir_recursive(dir_name) != OK:
-				failed_packages.add(package_name, "Unable to create directory")
-				continue
-		
-		res = xzf(download_location, dir_name)
-		if not res or res.is_err():
-			failed_packages.add(package_name, res.unwrap_err().to_string() if res else DEFAULT_ERROR)
-			continue
-		
-		if dir.remove(download_location) != OK:
-			failed_packages.add(package_name, "Failed to remove tarball")
-			continue
-
-		lock_file[package_name] = {
-			LockFileKeys.VERSION: package_version,
-			LockFileKeys.INTEGRITY: npm_manifest["dist"]["integrity"]
-		}
-
-		# TODO read package.json file of downloaded package and download dependencies
-		var file := File.new()
-		var npm_package_file_location := "%s/%s" % [dir_name, NPM_PACKAGE_FILE]
-		if not file.file_exists(npm_package_file_location):
-			failed_packages.add(package_name, "No npm package manifest found, unable to read metadata")
-			continue
-
-		if file.open(npm_package_file_location, File.READ) != OK:
-			failed_packages.add(package_name, "Unable to open npm package manifest, unable to read metadata")
-			continue
-		
-		res = _json_to_dict(file.get_as_text())
-		if not res or res.is_err:
-			failed_packages.add(package_name, "Unable to read npm package manifest")
-			continue
+	# Used for compiling together all errors that may occur
+	var failed_packages := FailedPackages.new()
+	for pkg in pkg_configs:
+		res = yield(pkg.download(), "completed")
+		if res.is_err():
+			failed_packages.add(pkg.name, str(res))
 
-		var npm_package_manifest: Dictionary = res.unwrap()
-		
-	
+	res = modify_packages()
+	if res.is_err():
+		printerr(res)
```

Shouldn't this return an error instead of continuing execution?

---

Oh yeah, I left that there.